### ISSUE TITLE
Cannot add Service Principal or Security Group to PowerBI Workspace

### ISSUE THREAD

**Issue by davec-fvr** [MicrosoftDocs/powerbi-docs#4337]
I've been following the steps here https://learn.microsoft.com/en-us/power-bi/developer/embedded/embed-service-principal and I cannot add a Service Principal or Security Group to PowerBI Workspace. I have done the following:

* Created an app in the portal in App Registrations
  * Created a secret key
* Created a security group in the portal
  * Added the app/service principal as a member of the security group
* Enabled Enable Embed content in apps in the PowerBI admin section
  * Tested with this with "The entire organisation" and "Specific security group" set to the security group above
  * Enabled Allow service principals to use Power BI APIs and specified the security group created above
It's this point where things fall down. The screenshots look out of date in the documentation, but I tried the following for step 4
* Settings -> Admin portal -> Workspaces
* Selected the ... menu next to the workspace I want to embed and chose Access
* In the email address field, attempted to type the name of my service principal or security group. Both displayed as invalid email addresses. It appears I can only add members in this field
* Tried the Powershell version from the docs and got the following:
```
Add-PowerBIWorkspaceUser: C:\Users\Dave Conley\add sp to workspace.ps1:11
Line |
  11 |  Add-PowerBIWorkspaceUser -Id $($pbiWorkspace.Id) -AccessRight Member  …
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Operation returned an invalid status code 'Forbidden'
```
* Tried using the API here https://learn.microsoft.com/en-us/rest/api/power-bi/groups/add-group-user#code-try-0 and got the following
```
{
  "error": {
    "code": "InvalidRequest",
    "message": "Failed to get service principal details from AAD."
  }
}
```
Clearly I've missed something but I can't find what. Any help would be appreciated.

**Comment by bishalgoswami** [comment]
Thanks for your comment, @davec-fvr. This issue has been assigned to the doc author for review.

**Comment by davec-fvr** [comment]
I think I found the problem, but I can't be sure as I've changed some other things in trying to solve this but it looks like 
1. Going to Settings->Admin Portal->Workspaces and selecting Access does not allow me to add a SP or security group as a member or admin user, however going to "Workspaces" and selecting my own workspace and pressing "Access" on the top navigation of that screen _does_ allow me to add a SP and SG even though the UI for both are as good as identical
2. The documentation tells me to create an App in App registrations but it doesn't tell me to add the Power BI Service API permissions. This is mentioned in a completely different set of documentation. 

These docs (Power BI in general, not just this specific page) are _really_ hard to follow. There is so much information and so many links to other documents. Add on top that some docs explain how to do the same thing in three different ways but these would be better displayed in tabs rather than one after each other in a step by step document. I realised I had created an app twice (once in portal and once in powershell) because I was reading top to bottom and thinking I had to do everything. It's also difficult to know which steps are required at all since some steps are for embed for org and some for embed for customer and some for both.

**Comment by KesemSharabi** [comment]
Hi @davec-fvr. 

Thanks for your valuable feedback and for getting involved. I'm happy that you found a solution to your problem.

We'll take into consideration your comments when creating future documents, and when we update the current ones.

Thanks,

Kesem
